syntax = "proto3";

package agora.chat_transcriber_python;
option go_package = ".;pb";

message Text {
  int32 vendor = 1;
  int32 version = 2;
  int32 seqnum = 3;
  int32 uid = 4;
  int32 flag = 5;
  int64 time = 6; // final time =first nofinal time
  int32 lang = 7;
  int32 starttime = 8;
  int32 offtime = 9;
  repeated Word words = 10;
  bool end_of_segment = 11;
  int32 duration_ms = 12;
  string data_type = 13; // transcribe ,translate
  repeated Translation trans = 14;
  string culture = 15;
  int64 texttime = 16; // pkg timestamp
}

message Word {
  string text = 1;
  int32 start_ms = 2;
  int32 duration_ms = 3;
  bool is_final = 4;
  double confidence = 5;
}

message Translation {
  bool is_final = 1;
  string lang = 2;
  repeated string texts = 3;
}